CMakeLists.txt

```diff
@@ -0,0 +1,69 @@
+message( STATUS "---------Beginning cmake build of ld-ldf-reader---------")
+
+cmake_minimum_required( VERSION 3.22 )
+project(ld-ldf-reader)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+set(CMAKE_CXX_STANDARD 17)
+
+set(CMAKE_AUTOUIC_SEARCH_PATHS "${SMAKE_CURRENT_SOURCE_DIR}/ui")
+
+# Include all the paths
+message( STATUS "--------- Finding packages---------")
+
+find_package(Qt5 REQUIRED COMPONENTS Core)
```

Comment:
I'd definitely get it working with Qt 5 first, but we will need Qt 6 support as well (see the DomesdayDuplicator commits for how we handled it there without needing to make too much stuff conditional).